### ISSUE TITLE
SPAWN_REASON blacklist does not work

### ISSUE THREAD

**Issue by homerek111** [BG-Software-LLC/WildStacker#966]
### Minecraft's Version

This server is running Paper version 1.21.1-128-master@d348cb8 (2024-10-21T16:23:24Z) (Implementing API version 1.21.1-R0.1-SNAPSHOT)

### Plugin's Version

2024.3-b86

### Describe the bug

I tried using:
```
  # Blacklisted entities are entities that won't get stacked.
  # EntityType list: https://bg-software.com/entities/
  # SpawnReason list: https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/event/entity/CreatureSpawnEvent.SpawnReason.html
  # You can combine both filters using "ENTITY_TYPE:SPAWN_REASON"
  # If you wish to disable blacklisted entities, use "blacklist: []"
  blacklist:
  - "ALL:SHOULDER_ENTITY"
  - "ALL:NATURAL"
  - "ALL:CUSTOM"
```
and i tried also:
```
  blacklist:
  - "SHOULDER_ENTITY"
  - "NATURAL"
  - "CUSTOM"
```

I also tried it on some specific mob type (enderman to easy check that with natural spawn in end)
```
  blacklist:
  - "ENDERMAN:SHOULDER_ENTITY"
  - "ENDERMAN:NATURAL"
  - "ENDERMAN:CUSTOM"
```

but it does not work at all and entities from these spawn reasons are still stacking

i tried it with 
```
  stack-checks:
    SPAWN_REASON: true
```
and ``false`` as well

### To Reproduce

-

**Comment by homerek111** [comment]
@OmerBenGera could you please take a look at that bug as it's really harmful for other plugins :/

**Comment by OmerBenGera** [comment]
Check out latest dev build, let me know if it is fixed.
This fix should work on paper servers 1.16+ - for spigot versions / 1.12 or below, open a new issue.

**Comment by homerek111** [comment]
> Check out latest dev build, let me know if it is fixed. This fix should work on paper servers 1.16+ - for spigot versions / 1.12 or below, open a new issue.

seems to be fixed
